pubsub/rabbitmq/metadata.yaml

```diff
@@ -0,0 +1,168 @@
+      example: '"password"'
+metadata:
+  - name: durable
+    type: bool
+    description: |
+      Whether or not to use durable queues.
+    default: '"false"'
+    example: '"true", "false"'
+  - name: publisherConfirm
+    type: bool
+    description: |
+      If enabled, client waits for publisher confirmation after publishing
+      a message.
+    default: '"false"'
+    example: '"true", "false"'
+  - name: maxLen
+    type: number
+    description: |
+      The maximum number of messages of a queue and its dead letter queue
+      (if dead letter enabled). If both maxLen and maxLenBytes are set then
+      both will apply; whichever limit is hit first will be enforced.
+    example: '1000'
+  - name: maxLenBytes
+    type: number
+    description: |
+      Maximum length in bytes of a queue and its dead letter queue (if dead
+      letter enabled). If both maxLen and maxLenBytes are set then both will 
+      apply; whichever limit is hit first will be enforced.
+    example: '1048576'
+  - name: concurrency
+    type: string
+    description: |
+      Allows processing multiple messages in
+      parallel (limited by the app-max-concurrency annotation, if configured).
+      Set to single to disable parallel processing. In most situations there’s 
+      no reason to change this.
+    example: '"parallel", "single"'
+    default: '"parallel"'
+  - name: enableDeadLetter
+    type: bool
+    description: |
+      Enable forwarding messages that cannot be handled to a dead-letter
+      topic.
+    default: '"false"'
+    example: '"true", "false"'
+  - name: prefetchCount
+    type: number
+    description: |
+      Number of messages to prefetch. Consider changing this to a non-zero
+      value for production environments. The value of "0" means that
+      all available messages will be pre-fetched.
+    default: '0'
+    example: '2'
+  - name: exchangeKind
+    type: string
+    description: |
+      Exchange kind of the rabbitmq exchange.
+    default: '"fanout"'
+    example: '"fanout","topic"'
+  - name: deliveryMode
+    type: number
+    description: |
+      Persistence mode when publishing messages.
+      RabbitMQ treats "2" as persistent, all other numbers as non-persistent.
+    default: '0'
+    example: '2'
+  - name: autoAck
+    type: bool
+    description: |
+      Whether or not the queue consumer should auto-ack messages.
+    default: '"false"'
+    example: '"true", "false"'
+  - name: deletedWhenUnused
+    type: bool
+    description: |
+      Whether or not the queue should be configured to auto-delete.
+    default: '"true"'
+    example: '"true", "false"'
+  - name: requeueInFailure
+    description: |
+      Whether or not to requeue when sending a negative acknowledgement
+      in case of a failure.
+    type: bool
+    default: '"false"'
+    example: '"true", "false"'
+  - name: reconnectWaitSeconds
+    description: |
+      Reconnect wait in Seconds.
+    type: duration
```

Comment:
Same comments as the ones @ItalyPaleAle  made for `ttlInSeconds`: 

1. this seems like a misnomer, the name induces a `number` but the type is `duration`?!
2. no default, 
3. no example

Reply:
This is not a misnomer - this is our standard implementation of durations. If you specify an integer without unit we interpret it as seconds! If you specify a duration string like "500ms" or "2m" we accept that too.

Reply:
Components should agree on metadata names. We should not have one as `ttlInSeconds` accepting numbers and another one with `ttlInSeconds` accepting duration and another one using only `ttl` with duration.

This PR is about documenting the existing attributes. I don't think we should block for that, instead, we should have a separate issue to move as many of the attributes to support duration - except when the underlying client does not support it. For example, if a TTL is always in seconds for a given message broker, duration might be odd because it cannot support ms and the behavior cannot be honored (it would need to be 1s or 0s).

Reply:
As explained in Discord - this is an internal convenience implementation due to standardizing of metadata parsing.

Originally this field used an integer type. I changed it to `time.duration`, knowing that the duration parses we used falls back to trying integers and interpreting them as seconds.

Ideally we can rename this field to be just `ttl`, but keep `ttlInSeconds` arounds for backwards compatibility.

For now it's fine to document this field as either `integer` or `duration` - it doesn't matter to me.